target_chains/solana/programs/pyth-solana-receiver/src/state.rs

```diff
@@ -1,25 +1,30 @@
 use {
+    crate::error::ReceiverError,
     anchor_lang::prelude::*,
     std::{
         io::Write,
         ops::Deref,
-        str::FromStr,
     },
-    wormhole_solana::VAA,
+    wormhole_anchor_sdk::wormhole::PostedVaaData,
 };
 
 // The current chain's wormhole bridge owns the VAA accounts
 impl Owner for AnchorVaa {
     fn owner() -> Pubkey {
-        // wormhole address on solana devnet
-        Pubkey::from_str("3u8hJUVTA4jH1wYAyUur7FFZVQ8H635K3tSHHF4ssjQ5").unwrap()
+        PostedVaaData::owner()
     }
 }
 
 impl AccountDeserialize for AnchorVaa {
     // Manual implementation because this account does not have an anchor discriminator
     fn try_deserialize(buf: &mut &[u8]) -> anchor_lang::Result<Self> {
-        Self::try_deserialize_unchecked(buf)
+        require!(buf.len() >= 3, ErrorCode::AccountDiscriminatorNotFound);
+        let given_disc = &buf[..3];
+        require!(
+            *given_disc == *b"vaa",
+            ReceiverError::PostedVaaHeaderWrongMagicNumber
+        );
+        Self::try_deserialize_unchecked(&mut &buf[3..])
```

Comment:
This code doesn't make sense. The unchecked method should get the same buffer as the checked method -- it's just supposed to ignore the discriminator.